modules/core/src/main/java/org/apache/ignite/internal/processors/cache/IgniteCacheProxyImpl.java

```diff
@@ -553,8 +553,11 @@ private ClusterGroup projection(boolean loc) {
         if (qry instanceof ContinuousQuery) {
             ContinuousQuery<K, V> qry0 = (ContinuousQuery<K, V>)qry;
 
-            if (qry0.getLocalListener() == null)
-                throw new IgniteException("Mandatory local listener is not set for the query: " + qry);
+            if (qry0.getLocalListener() == null &&
+                qry0.getRemoteFilterFactory() == null && qry0.getRemoteFilter() == null) {
```

Comment:
Is this situation possible: qry0.getRemoteFilter() is not null, but qry0.getRemoteFilterFactory() returns null?